admin_server/src/main/java/com/backend/admin_server/access_requests/service/AccessRequestValidationService.java

```diff
@@ -118,6 +121,35 @@ private boolean sendForExternalVerification(String clientBase64Image, String use
         }
     }
 
+    private void validateBase64Images(String clientImage, String userImage) {
+        boolean clientImageValid = isValidBase64(clientImage);
+        boolean userImageValid = isValidBase64(userImage);
+
+        if (!clientImageValid || !userImageValid) {
+            StringBuilder invalidMessage = new StringBuilder("Invalid Base64 image string(s):");
+            if (!clientImageValid) {
+                invalidMessage.append(" Client Image");
+            }
+            if (!userImageValid) {
+                invalidMessage.append(" User Image");
+            }
+            LOGGER.severe(invalidMessage.toString());
+            throw new IllegalArgumentException(invalidMessage.toString());
+        }
+    }
+
+    private boolean isValidBase64(String base64) {
+        if (base64 == null || base64.isEmpty()) {
+            return false;
+        }
+        try {
+            Base64.getDecoder().decode(base64);
+            return true;
+        } catch (IllegalArgumentException e) {
```

Comment:
Generally speaking I am not a fan of using exceptions for control flow like this, but here it think it is ok as I don't see much other option.

Reply:
I agree, if the base64 string is not valid, the way we have it, it will keep the method running. It'd be better to throw exception and then end the execution because we won't be able to use that image at all, or am I missing something?